src/ecs/commands/runCommandInContainer.ts

```diff
@@ -0,0 +1,162 @@
+                'AWS.command.ecs.runCommandInContainer.noTasks',
+                'There are no running tasks for the service: {0}',
+                node.serviceName
+            )
+        )
+        return
+    }
+
+    const quickPick = picker.createQuickPick({
+        options: {
+            title: localize('AWS.command.ecs.runCommandInContainer.chooseTask', 'Choose a task'),
+            ignoreFocusOut: true,
+        },
+        items: quickPickItems,
+    })
+
+    let taskChoice
+
+    if (quickPickItems.length === 1) {
+        taskChoice = quickPickItems
+    } else {
+        taskChoice = await picker.promptUser({
+            picker: quickPick,
+            onDidTriggerButton: (button, resolve, reject) => {
+                if (button === vscode.QuickInputButtons.Back) {
+                    resolve(undefined)
+                }
+            },
+        })
+    }
+
+    if (!taskChoice) {
+        return
+    }
+
+    const task = taskChoice[0].label
+
+    //Get command line text from user to run in the container
+    const command = await window.showInputBox({
+        prompt: localize(
+            'AWS.command.ecs.runCommandInContainer.prompt',
+            'Enter the command to run in container: {0}',
+            node.continerName
+        ),
+        placeHolder: localize('AWS.command.ecs.runCommandInContainer.placeHolder', 'Command to run'),
+        ignoreFocusOut: true,
+    })
+    if (!command) {
+        return
+    }
+
+    // Warn the user that commands may modify the container, give the option to dismiss future prompts
+    const configuration = new DefaultSettingsConfiguration(extensionSettingsPrefix)
+    if (configuration.readSetting('ecs.warnRunCommand')) {
+        const choice = await window.showInformationMessage(
+            localize(
+                'AWS.command.ecs.runCommandInContainer.warnBeforeExecute',
+                'Command may modify the running container {0}. Are you sure?',
+                node.continerName
+            ),
+            { modal: true },
+            localize('AWS.generic.response.yes', 'Yes'),
+            localize('AWS.message.prompt.yesDontAskAgain', "Yes, and don't ask again")
+        )
+        if (choice === undefined) {
+            getLogger().debug('ecs: Cancelled runCommandInContainer')
+            return
+        } else if (choice === localize('AWS.message.prompt.yesDontAskAgain', "Yes, and don't ask again")) {
+            configuration.writeSetting('ecs.warnRunCommand', false, vscode.ConfigurationTarget.Global)
+        }
+    }
+
+    const response = await new ChildProcess(
+        true,
+        'aws',
+        undefined,
+        'ecs',
+        'execute-command',
+        '--cluster',
+        node.clusterArn,
+        '--task',
+        task,
+        '--container',
+        node.continerName,
+        '--command',
+        command,
+        '--interactive'
+    ).run()
+
+    if (response.exitCode !== 0) {
+        getLogger().error(response.stderr)
```

Comment:
- need some sort of prefix, otherwise this is unreadable in the logs. 
- storing ChildProcess as a local var allows us to get the string representation of it, for logging.

```suggestion
        const cmd = await new ChildProcess(...)
        const response = cmd.run()
        getLogger().error('ECS command failed: %O: %O', cmd, response.stderr)
```

Reply:
That's really useful, thanks.